File: spec/lib/fizzbuzz_spec.rb
require 'spec_helper'
require 'fizzbuzz'

describe Game do
  context "when passing 3" do
    it "returns Fizz" do
      expect(subject.check(3)).to eq("Fizz")
    end
  end
  
  context "when passing 5" do
    it "returns Buzz" do
      expect(subject.check(5)).to eq("Buzz")
    end
  end
  
  context "when passing 15" do
    it "returns FizzBuzz" do
      expect(subject.check(15)).to eq("FizzBuzz")
    end
  end
  
  context "when passing 4" do
    it "returns 4" do
      expect(subject.check(4)).to eq(4)
    end
  end
end

File: javascript/spec/FizzBuzzSpec.js
describe("Fizz", function() {
  var fizz;

  beforeEach(function() {
    fizz = new Fizz(3);
  });

  it("should return true if the number is divisible by 3", function() {
    
    expect(fizz.is()).toEqual(true);
  });
});

describe("Buzz", function() {
  var buzz;

  beforeEach(function() {
    buzz = new Buzz(5);
  });

  it("should return true if the number is divisible by 5", function() {
    
    expect(buzz.is()).toEqual(true);
  });
});

describe("FizzBuzz", function() {
  var fizzbuzz;

  beforeEach(function() {
    fizzbuzz = new FizzBuzz(15);
  });

  it("should return true if the number is divisible by 15", function() {
    
    expect(fizzbuzz.is()).toEqual(true);
  });
});

describe("Game", function() {
  var game;

  beforeEach(function() {
    game = new Game(15);
  });

  it("should return fizzbuzz for 15", function() {
    
    expect(game.run()).toEqual("fizzbuzz");
  });
});
